Dashboard/app/js/lib/views/views-public.js

```diff
@@ -67,6 +139,39 @@ Ember.Handlebars.registerHelper('placemarkDetail', function () {
   return new Handlebars.SafeString(markup);
 });
 
+/*  Take a timestamp and render it as a date in format
+    YYYY-mm-dd */
+function renderTimeStamp(timestamp) {
+  var d, t, date, month, year;
+  t = parseInt(timestamp, 10);
+  if (isNaN(t)) {
+    return "";
+  }
+
+  d = new Date(t);
+  if (d) {
```

Comment:
Instead of having a big `if` with an small `else` you can just `return ""` if `!d`